packages/api-client/src/auth/AccessTokenStore.ts

```diff
@@ -30,11 +30,19 @@ export interface AccessTokenStore {
   on(event: TOPIC.ACCESS_TOKEN_REFRESH, listener: (accessToken: AccessTokenData) => void): this;
 }
 
+type AccessTokenWithExpirationDate = AccessTokenData & {
+  /**
+   * The date at which the token will be invalid. This value should be use with a grain of salt as there might be some time shift between browser and server time
+   * It is suggested to add an error margin
+   */
+  expireAt: number;
```

Comment:
shouldn't this be included in the access token interface itself?

Reply:
Well the `AccessTokenData` is the raw response from backend (which doesn't include the expireAt), we compute it only here, so I think it makes sense to have a new type on top of `AccessTokenData`. WDYT?

Reply:
But maybe you are right, I should not add it to the same object but rather another property of the store, let me try this

Reply:
whats the difference between expiresIn / expiresAt

Reply:
expiresIn is the time that the token is valid (15min)
expiresAt is the Date at which the token will be invalid (June 15th 18:35 for example)

expiresAt is computed from expiresIn